rm-datasource/src/main/java/io/seata/rm/datasource/undo/AbstractUndoLogManager.java

```diff
@@ -431,4 +433,19 @@ protected abstract void insertUndoLogWithNormal(String xid, long branchId, Strin
     protected boolean needCompress(byte[] undoLogContent) {
         return ROLLBACK_INFO_COMPRESS_ENABLE && undoLogContent.length > ROLLBACK_INFO_COMPRESS_THRESHOLD;
     }
+
+    @Override
+    public boolean hasUndoLogTable(Connection conn) {
+        String checkExistSql = getCheckUndoLogTableExistSql();
+        try (PreparedStatement checkPst = conn.prepareStatement(checkExistSql)) {
+            checkPst.executeQuery();
+            return true;
+        } catch (SQLException e) {
+            return false;
+        }
+    }
```

Comment:
How about using the following SQL statement instead of throwing an exception to determine whether a table exists?
```
SELECT * 
FROM information_schema.tables
WHERE table_schema = 'db_name' 
    AND table_name = 'table_name'
LIMIT 1
```

Reply:
同上